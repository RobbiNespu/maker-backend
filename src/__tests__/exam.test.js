const { gqlTestClient } = require('./utils/gqlTestClient')
const { prisma } = require('./utils/createPrismaMock')
const { userMock, coverMock } = require('./mocks')

describe('Exam Resolvers', () => {
  test('Exam owned can get exam by ID', async () => {
    const response = await gqlTestClient(
      /* GraphQL */ `
        {
          exam(id: "cjssjhye6de2t0b706ofy7552") {
            id
            title
            description
            pass
            time
            downloads
            published
            createdAt
            cover {
              id
              variant
              text
            }
          }
        }
      `,
      null,
      userMock()
    )
    expect(prisma.exam).toHaveBeenCalledWith({ id: 'cjssjhye6de2t0b706ofy7552' })
    expect(userMock).toHaveBeenCalled()
    expect(coverMock).toHaveBeenCalled()
    expect(response).toMatchSnapshot()
  })
})
