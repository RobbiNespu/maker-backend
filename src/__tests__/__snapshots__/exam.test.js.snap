// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exam Resolvers Get Exam By ID 1`] = `
Object {
  "data": Object {
    "exam": Object {
      "cover": Array [
        Object {
          "id": "cjssjhyeede2w0b70npid97is",
          "text": "UFC Quiz",
          "variant": 2,
        },
        Object {
          "id": "cjssjhyeede2w0b70npid97it",
          "text": "How well do you know your mixed martial arts ?",
          "variant": 1,
        },
      ],
      "createdAt": "2019-03-03T06:33:46.589Z",
      "description": "Test your knowledge of mixed martial arts.",
      "downloads": 8,
      "id": "cjssjhye6de2t0b706ofy7552",
      "pass": 65,
      "published": true,
      "time": 10,
      "title": "UFC Quiz",
    },
  },
}
`;
